testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/cluster/AbstractFailoverClusterTest.java

```diff
@@ -67,45 +118,40 @@ protected void switchFailedNode() {
         assertFalse(controller.isStarted(getCurrentFailNode().getQualifier()));
     }
 
-    protected Cookie login(AbstractPage targetPage) {
-        targetPage.navigateTo();
-        assertCurrentUrlStartsWith(loginPage);
-        loginPage.form().login(ADMIN, ADMIN);
-        assertCurrentUrlStartsWith(targetPage);
+    protected Cookie login() {
+        oauth.openLoginForm();
+        assertTrue(loginPage.isCurrent());
+        loginPage.login("test-user@localhost", "password");
+        assertTrue(appPage.isCurrent());
         Cookie sessionCookie = driver.manage().getCookieNamed(KEYCLOAK_SESSION_COOKIE);
         assertNotNull(sessionCookie);
         return sessionCookie;
     }
 
-    protected void logout(AbstractPage targetPage) {
-        if (!(targetPage instanceof PageWithLogOutAction)) {
-            throw new IllegalArgumentException(targetPage.getClass().getSimpleName() + " must implement PageWithLogOutAction interface");
-        }
-        targetPage.navigateTo();
-        assertCurrentUrlStartsWith(targetPage);
-        ((PageWithLogOutAction) targetPage).logOut();
+    protected void logout() {
+        appPage.logout();
     }
 
-    protected Cookie verifyLoggedIn(AbstractPage targetPage, Cookie sessionCookieForVerification) {
+    protected Cookie verifyLoggedIn(Cookie sessionCookieForVerification) {
         // verify on realm path
-        masterRealmPage.navigateTo();
+        URLUtils.navigateToUri(AUTH_SERVER_ROOT + "/realms/test");
         Cookie sessionCookieOnRealmPath = driver.manage().getCookieNamed(KEYCLOAK_SESSION_COOKIE);
         assertNotNull(sessionCookieOnRealmPath);
         assertEquals(sessionCookieOnRealmPath.getValue(), sessionCookieForVerification.getValue());
         // verify on target page
-        targetPage.navigateTo();
-        assertCurrentUrlStartsWith(targetPage);
+        appPage.open();
+        assertTrue(appPage.isCurrent());
         Cookie sessionCookie = driver.manage().getCookieNamed(KEYCLOAK_SESSION_COOKIE);
         assertNotNull(sessionCookie);
         assertEquals(sessionCookie.getValue(), sessionCookieForVerification.getValue());
         return sessionCookie;
     }
 
-    protected void verifyLoggedOut(AbstractPage targetPage) {
+    protected void verifyLoggedOut() {
         // verify on target page
-        targetPage.navigateTo();
+        oauth.openLoginForm();
         driver.navigate().refresh();
-        assertCurrentUrlStartsWith(loginPage);
+        assertTrue(loginPage.isCurrent());
```

Comment:
Could be here and also in the other places for better error reporting:
```suggestion
        loginPage.assertCurrent();
```